src/bsoncxx/oid.cpp

```diff
@@ -23,31 +25,32 @@
 namespace bsoncxx {
 BSONCXX_INLINE_NAMESPACE_BEGIN
 
-const oid::init_tag_t oid::init_tag{};
-
-oid::oid() : _is_valid(false) {
-}
-
-oid::oid(init_tag_t) : _is_valid(true) {
+oid::oid() {
     bson_oid_t oid;
     bson_oid_init(&oid, nullptr);
 
     std::memcpy(_bytes.data(), oid.bytes, sizeof(oid.bytes));
 }
 
-oid::oid(const string::view_or_value& str)
-    : _is_valid(bson_oid_is_valid(str.data(), str.view().length())) {
-    if (_is_valid) {
-        bson_oid_t oid;
-        bson_oid_init_from_string(&oid, str.terminated().data());
-        memcpy(_bytes.data(), oid.bytes, _bytes.size());
+const oid::init_tag_t oid::init_tag{};
+
+oid::oid(init_tag_t) : oid::oid() {
+}
+
+oid::oid(const bsoncxx::stdx::string_view& str) {
+    if (!bson_oid_is_valid(str.data(), str.size())) {
+        throw bsoncxx::exception{error_code::k_invalid_bson};
```

Comment:
This actually makes more sense, I'll just rename `k_invalid_bson` to `k_invalid_oid` since we're not using `k_invalid_bson` anywhere.